spread-shellcheck

```diff
@@ -133,15 +137,59 @@ def checksection(data, env: Dict[str, str]):
                             stdout=subprocess.PIPE,
                             stdin=subprocess.PIPE,
                             shell=True)
-    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=30)
+    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=60)
     if proc.returncode != 0:
         raise ShellcheckRunError(stdout)
 
 
+class Cacher:
+    _instance = None
+    def __new__(cls):
+        if cls._instance is not None:
+            return cls._instance
+        instance = super().__new__(cls)
+        cls._instance = instance
+        instance.enabled = True
+        return instance
+
+    def disable(self):
+        logging.debug("caching is disabled")
+        self.enabled = False
```

Comment:
We could delete the cache here and take the opportunity to rebuild it with just the necessary files, since we're already running every check anyway. Since the cache accumulates files that eventually become unnecessary due to tests changing. The cache is small enough so that it's not a big problem but this way we'd rebuild "for free". We'd also have to remove the early check in `cache_success` though

Reply:
It was kind of my initial idea, hence is_cached returned the digest, but then I abandoned it. I'd rather have a separate --rebuild-cache flag. With the latest tweaks I pushed, with --no-cache the cache operations are supposed to be nops.